### PR TITLE
feat: PRT-adding-api-usage-metrics

### PR DESCRIPTION
**Pull request by ranlavanet**

### CODE REVIEW

File: protocol/metrics/metrics_consumer_manager.go [hunk 1]
```diff
@@ -28,14 +28,16 @@ type ConsumerMetricsManager struct {
...
 	endpointsHealthChecksOkMetric          prometheus.Gauge
 	endpointsHealthChecksOk                uint64
 	lock                                   sync.Mutex
 	protocolVersionMetric                  *prometheus.GaugeVec
 	providerRelays                         map[string]uint64
+	addMethodsApiGauge                     bool
 }
 
-func NewConsumerMetricsManager(networkAddress string) *ConsumerMetricsManager {
+func NewConsumerMetricsManager(networkAddress string, addMethodsApiGauge bool) *ConsumerMetricsManager {
```
Comment (shleikes) [inline protocol/metrics/metrics_consumer_manager.go:40]: I don't like that this option is transferred like this. It should either be a global flag, being set from Cobra, or an options struct, with flags like this one, which will be probably be added in the future

File: protocol/metrics/metrics_consumer_manager.go [hunk 2]
```diff
@@ -28,14 +28,16 @@ type ConsumerMetricsManager struct {
 	LatestBlockMetric                      *prometheus.GaugeVec
 	LatestProviderRelay                    *prometheus.GaugeVec
 	virtualEpochMetric                     *prometheus.GaugeVec
+	apiSpecificsMetric                     *prometheus.GaugeVec
```
Comment (shleikes) [inline protocol/metrics/metrics_consumer_manager.go:31]: Why not `apiMethodCalls`? You've already called the new field in `RelayMetrics`: `ApiMethod`

File: protocol/metrics/metrics_consumer_manager.go [hunk 3]
```diff
@@ -202,6 +212,9 @@ func (pme *ConsumerMetricsManager) SetRelayMetrics(relayMetric *RelayMetrics, er
 	pme.latencyMetric.WithLabelValues(relayMetric.ChainID, relayMetric.APIType).Set(float64(relayMetric.Latency))
 	pme.totalCURequestedMetric.WithLabelValues(relayMetric.ChainID, relayMetric.APIType).Add(float64(relayMetric.ComputeUnits))
 	pme.totalRelaysRequestedMetric.WithLabelValues(relayMetric.ChainID, relayMetric.APIType).Add(1)
+	if pme.addMethodsApiGauge && relayMetric.ApiMethod != "" { // pme.addMethodsApiGauge never changes so its safe to read concurrently
+		pme.apiSpecificsMetric.WithLabelValues(relayMetric.ChainID, relayMetric.APIType, relayMetric.ApiMethod).Add(1)
```
Comment (shleikes) [inline protocol/metrics/metrics_consumer_manager.go:216]: What about Connection Type?

File: protocol/metrics/metrics_provider_manager.go [hunk 4]
```diff
@@ -14,9 +14,10 @@ import (
 )
 
 const (
-	MetricsListenFlagName = "metrics-listen-address"
-	RelayServerFlagName   = "relay-server-address"
-	DisabledFlagOption    = "disabled"
+	MetricsListenFlagName       = "metrics-listen-address"
+	AddApiMethodMetricsFlagName = "add-api-method-metrics"
```
Comment (shleikes) [inline protocol/metrics/metrics_provider_manager.go:18]:
```suggestion
	AddApiMethodCallsMetricsFlagName = "add-api-method-metrics"
```

File: protocol/rpcconsumer/rpcconsumer.go [hunk 5]
```diff
@@ -98,9 +98,10 @@ type ConsumerStateTrackerInf interface {
 }
 
 type AnalyticsServerAddressess struct {
-	MetricsListenAddress string
-	RelayServerAddress   string
-	ReportsAddressFlag   string
+	AddApiMethodMetricsFlagName bool
```
Comment (shleikes) [inline protocol/rpcconsumer/rpcconsumer.go:101]:
```suggestion
	AddApiMethodCallsMetricsFlagName bool
```

File: scripts/pre_setups/init_lava_only_with_node.sh [hunk 6]
```diff
@@ -57,7 +57,7 @@ wait_next_block
 
 screen -d -m -S consumers bash -c "source ~/.bashrc; lavap rpcconsumer \
 127.0.0.1:3360 LAV1 rest 127.0.0.1:3361 LAV1 tendermintrpc 127.0.0.1:3362 LAV1 grpc \
-$EXTRA_PORTAL_FLAGS --geolocation 1 --log_level debug --from user1 --chain-id lava --pprof-address "127.0.0.1:6060"  --allow-insecure-provider-dialing --metrics-listen-address ":7779" 2>&1 | tee $LOGS_DIR/CONSUMERS.log" && sleep 0.25
+$EXTRA_PORTAL_FLAGS --geolocation 1 --log_level debug --from user1 --chain-id lava --add-api-method-metrics  --allow-insecure-provider-dialing --metrics-listen-address ":7779" 2>&1 | tee $LOGS_DIR/CONSUMERS.log" && sleep 0.25
```
Comment (shleikes) [inline scripts/pre_setups/init_lava_only_with_node.sh:60]: Why did you remove the pprof flag?

File: x/protocol/types/params.go [hunk 7]
```diff
@@ -12,7 +12,7 @@ import (
 var _ paramtypes.ParamSet = (*Params)(nil)
 
 const (
-	TARGET_VERSION = "1.2.4"
+	TARGET_VERSION = "1.2.5"
```
Comment (shleikes) [inline x/protocol/types/params.go:15]: Separate PR

**Review by shleikes — CHANGES_REQUESTED** [review]

File: protocol/metrics/metrics_consumer_manager.go (hunk 3 above)
Comment (ranlavanet) [reply protocol/metrics/metrics_consumer_manager.go:216]: we dont care about connection type.

File: scripts/pre_setups/init_lava_only_with_node.sh (hunk 6 above)
Comment (ranlavanet) [reply scripts/pre_setups/init_lava_only_with_node.sh:60]: I dont want it to take a port :)

**Review by shleikes — APPROVED** [review]
